creator/ingest_runs/tasks.py

```diff
@@ -1,13 +1,109 @@
-def run_ingest(ingest_run_id):
-    # TODO
-    print(f"Preparing ingest run {ingest_run_id}")
+import django_rq
+import logging
 
+from django.core.exceptions import ValidationError
 
-def cancel_ingest(ingest_run_id):
-    # TODO
-    print(f"Cancelling ingest run {ingest_run_id}")
+from creator.decorators import task
+from creator.files.models import FileType
+from creator.ingest_runs.models import IngestRun
+from creator.utils import stop_job
+
+
+logger = logging.getLogger(__name__)
+
+
+@task("run_ingest")
+def run_ingest(ingest_run_id=None):
+    """
+    Run ingest process for a given set of files.
+    """
+    ingest_run = get_ingest_run(ingest_run_id)
+    logging.info(f"Preparing ingest run {ingest_run.pk} for processing.")
+
+    # Check if any IngestRuns with the same input hash exist. If so, cancel
+    # them and delete the newly created IngestRun, which never gets started.
+    # If not, start a new IngestRun.
+    if cancel_duplicate_ingest_runs(ingest_run):
```

Comment:
Hmm yea, maybe this check should happen in start ingest run mutation. We can check if an ingest run with the input hash exists there. If yes, we queue cancel_ingest task. If no, we create new ingest run and queue run_ingest task (what we have now). @gio thoughts?

Reply:
I wish that was my github name...

Reply:
Oo oops! Wrong person lol. 

I think Dan was just talking about the call to ingest_run.delete()? We shouldn't create a new one if we know we are going to have to delete it. I think we still want to cancel the existing ingest run so that we can achieve the "start, stop, start" behavior when someone pushes the ingest button 3 times.

Reply:
What I meant is, if we have a check in the start ingest run mutation for an ingest run with the same input hash, then we can't possibly have any ingest runs with duplicate input hashes.

Reply:
I think its still needed though right? Bc thats the only place where an ingest run is canceled inside the cancel_ingest task. I guess it should just be called `cancel_prev_ingest_run`

Reply:
That sounds right to me.
Do you only care about unique ingest runs in the non-terminal states, tho?
With the way things are now, how would someone re-running an IngestRun work?

Reply:
I just realized, if we have the start ingest run mutation calls the `cancel_ingest` task as needed, what's the purpose of the cancel ingest run mutation? As of now, it's not being used, right?

Reply:
Yea for now there is no purpose. But I think later on as the UI evolves, the start ingest mutation would only start the ingest and some other button on the UI would call the cancel ingest mutation to stop the run

Reply:
As an update, I've done this, but now the unit tests for this need to be rewritten